examples/echo/main.c

```diff
@@ -166,9 +185,19 @@ main (gint   argc,
     exit (1);
   }
 
-  GMainLoop *loop = g_main_loop_new (NULL, FALSE);
+  g_debug ("starting main loop of echo worker");
+  GMainLoop *loop = g_main_loop_new (main_context, FALSE);
+  g_source_set_callback (sigint_source, G_SOURCE_FUNC(sigint_callback), loop, NULL);
   g_main_loop_run (loop);
 
+  g_debug ("terminating echo worker");
+  g_object_unref (worker);
```

Comment:
Similar to #7, I think these can be marked as `g_autoptr` and the macro will take care to unref and free the objects before they go out of scope. Is there a reason why you explicitly unref them?